bundles/org.openhab.binding.ipcamera/README.md

```diff
@@ -0,0 +1,1312 @@
+
+
+## Supported Things
+
+If using openHAB's textual configuration or when needing to setup HABPANEL/sitemaps, you are going to need to know what your camera is as a "thing type". 
+These are listed in CAPS below. Example: The thing type for a generic onvif camera is "ONVIF".
+
+| Thing Type ID | Description |
+|-|-|
+| `AMCREST` | Use for all Amcrest Cameras that do not work as a Dahua thing. This uses an older polling method for alarm detection which is not as efficient as the newer method used in Dahua. Amcrest are made by Dahua and hence their cameras can be setup as a Dahua thing. |
+| `DAHUA` | Use for all current Dahua and Amcrest cameras as they support an API as well as ONVIF. |
+| `DOORBIRD` | Use for all current DOORBIRD cameras as they support an API as well as ONVIF. |
+| `FOSCAM` | Use for all current FOSCAM HD Cameras as they support an API as well as ONVIF. |
+| `HIKVISION` | Use for all current HIKVISION Cameras as they support an API as well as ONVIF. |
+| `HTTPONLY` | For any camera that is not ONVIF compatible yet still has the ability to fetch a snapshot or stream with a url. RTSP cameras should use this if they do not have ONVIF. |
+| `INSTAR` | Use for all current INSTAR Cameras as they support an API as well as ONVIF. |
+| `ONVIF` | Use for all ONVIF Cameras from any brand that does not have an API. |
+| `GROUPDISPLAY` | Used to display or cast multiple cameras like they are a single camera. This is an advanced feature that may require some tweaking of the cameras settings to fully work. |
+
+
+## Binding Configuration
+
+To add a camera just press on the PLUS (+) icon when in the INBOX, then select the IpCamera binding (binding must be installed first).
+The binding will auto search for ONVIF cameras so give it a minute to search before moving onto manually adding a camera, as the auto detection will fill in some port settings for you making life easier.
+After the camera is added you probably need to edit the camera by pressing on the pencil icon to add your USER/PASS.
+
+For any urls you override, leave the `user:pass@` out of the url as the binding will handle this for you and hide your password with * so it can not be read. This will also make changing your password much easier if it is only located in 1 field.
+
+Below is a list of the configuration parameters that can be used in textual configuration. 
+If you do not specify any of these, the binding will use the default which should work in most cases. 
+Very few of them are needed in order to get a working camera and examples are shown in the `Full Example` section.
+
+DO NOT try and change a setting using PaperUI after using textual configuration as the two will conflict as the text file locks the settings preventing them from changing. 
+
+
+| Parameter | Description |
+|-|-|
+| `IPADDRESS`| The IP address of your camera or NVR. You can also use Hostnames if your camera is not locked to a set IP. |
```

Comment:
Configuration parameters should be camelCase or snake_case.

Reply:
Acknowledged. So should I change this and create breaking changes for an unknown amount of users?

Reply:
Yes, please. A binding should only be considered stable when merged to the official repo - until then, the prio should be to have it aligned with the other official bindings as much as possible.

Reply:
OK, that will take a few days to make the changes which includes a huge change to the readme and then testing and proof reading. Will pick the low hanging fruit first so I can test the numerous other changes suggested in parallel.

Reply:
DONE